### PR TITLE
[JUJU-3352] Find oracle shapes values, rather than hardcode

### PR DESCRIPTION
**Pull request by hmlanigan**
First update the oci-go-sdk to version 65, the most recent.

Gather Shapes from the sdk rather than use hardcoded data in juju. This solution is not foolproof as some data must be inferred from strings with no types which is suboptimal. Make best effort to use types were available and parse strings where not. We should be 

Improve the data available for hardware characteristics including adding memory, cpus, az and real arch.

A number of questions arose during the course of this change, added TODOs to look in the future.
## QA steps

There should be no change to current functionality using the oracle cloud.

```sh
# setup and bootstrap the oracle cloud
$ juju deploy ubuntu

# when finished deploying check the instance data
juju:PRIMARY> db.instanceData.find().pretty()
{
	"_id" : "123e8ef9-6165-480e-8c11-78e943006f8a:0",
	"machineid" : "0",
	"instanceid" : "ocid1.instance.oc1.phx.anyhqljtniwq6sycokw5ragfmvqqct3lyf6dimbmn7vqjj3rcftn5qn7nx4q",
	"display-name" : "juju-006f8a-0",
	"model-uuid" : "123e8ef9-6165-480e-8c11-78e943006f8a",
	"arch" : "amd64",
	"mem" : NumberLong(7168),
	"rootdisk" : NumberLong(0),
	"cpucores" : NumberLong(1),
	"txn-revno" : 2
}
....
```

## Bug reference
https://bugs.launchpad.net/juju/+bug/1980006

### CODE REVIEW

File: provider/oci/images.go [hunk 1]
```diff
@@ -283,26 +299,107 @@ func instanceTypes(cli ComputeClient, compartmentID, imageID *string) ([]instanc
...
+	case strings.HasPrefix(shape, "BM."):
+		return string(BareMetal)
+	default:
+		return ""
+	}
+}
+
+var oracleAmdBm = fmt.Sprintf("%s|%s|%s|%s", string(ociCore.ShapePlatformConfigOptionsTypeAmdMilanBm), string(ociCore.ShapePlatformConfigOptionsTypeAmdRomeBm), string(ociCore.ShapePlatformConfigOptionsTypeIntelSkylakeBm), string(ociCore.ShapePlatformConfigOptionsTypeIntelIcelakeBm))
+var oracleAmdBmGpu = fmt.Sprintf("%s|%s", string(ociCore.ShapePlatformConfigOptionsTypeAmdMilanBmGpu), string(ociCore.ShapePlatformConfigOptionsTypeAmdRomeBmGpu))
+var oracleAmd = fmt.Sprintf("%s|%s", string(ociCore.ShapePlatformConfigOptionsTypeAmdVm), string(ociCore.ShapePlatformConfigOptionsTypeIntelVm))
```
Comment (SimonRichardson) [inline provider/oci/images.go:376]: nit: Can we var group these to make it easier to read.

```
var (
     ...
)
```

File: provider/oci/images.go [hunk 2]
```diff
@@ -283,26 +299,107 @@ func instanceTypes(cli ComputeClient, compartmentID, imageID *string) ([]instanc
...
+// oracle architectures and instance types to juju
+// values
+var archREs = []struct {
+	*regexp.Regexp
+	arch         string
+	instanceType string
+}{
+	{regexp.MustCompile(oracleAmd), arch.AMD64, string(VirtualMachine)},
+	{regexp.MustCompile(oracleAmdBm), arch.AMD64, string(BareMetal)},
+	{regexp.MustCompile(oracleAmdBmGpu), arch.AMD64, string(GPUMachine)},
```
Comment (SimonRichardson) [inline provider/oci/images.go:388]: Can we add a `String` method on the `InstanceType`, that way we don't need to cast every time? Also if we ever want to change the underlying implementation, we don't have to change everywhere.

**Review by SimonRichardson — APPROVED** [review]

**Comment by hmlanigan** [comment]
intermittent fails in worker/dbaccessor

**Pull request merged**